algosdk/atomic_transaction_composer.py

```diff
@@ -288,22 +307,24 @@ def add_method_call(
                 txn_list.append(method_args[i])
             else:
                 if abi.is_abi_reference_type(arg.type):
-                    current_type = abi.UintType(8)
+                    current_type: str | abi.ABIType = abi.UintType(8)
                     if arg.type == abi.ABIReferenceType.ACCOUNT:
                         address_type = AddressType()
                         account_arg = address_type.decode(
-                            address_type.encode(method_args[i])
+                            address_type.encode(
+                                cast(Union[str, bytes], method_args[i])
+                            )
                         )
-                        current_arg = populate_foreign_array(
+                        current_arg: Any = populate_foreign_array(
```

Comment:
Ideally this should be a stricter type but it cannot be made any stricter until the parameter type for `method_args` is constrained to not include `Any`. To make that argument more specific would likely require specifying types in other places throughout the `abi` package and is therefore outside the scope of this PR.